app/services/feature_flag.rb

```diff
@@ -65,12 +68,30 @@ def self.deactivate(feature_name)
   def self.active?(feature_name)
     raise unless feature_name.in?(FEATURES)
 
-    FEATURES[feature_name].feature.active?
+    feature_statuses[feature_name].presence || false
   end
 
   def self.sync_with_database(feature_name, active)
     feature = Feature.find_or_initialize_by(name: feature_name)
     feature.active = active
+
+    clear_cache!
     feature.save!
   end
+
+  def self.feature_statuses
+    Rails.cache.fetch(cache_key, expires_in: CACHE_EXPIRES_IN) do
+      Feature.where(name: FEATURES.keys).each_with_object({}) do |feature, hash|
```

Comment:
Could be lighter: `Feature.all.pluck(:name, :active).to_h.with_indifferent_access`? And we have to scope to `FEATURES.keys`?

Reply:
good shout will amend